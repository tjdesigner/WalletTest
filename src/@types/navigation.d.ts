/**
 * Learn more about using TypeScript with React Navigation:
 * https://reactnavigation.org/docs/typescript/
 */

import { ParamListBase } from "@react-navigation/native"
import { NativeStackScreenProps } from "@react-navigation/native-stack"

import type {
  CompositeScreenProps,
  NavigatorScreenParams,
} from "@react-navigation/native"
import type { StackScreenProps } from "@react-navigation/stack"

interface Card {
  id?: string
  cardNumber: string
  name: string
  expirationDate: string
  cvv: string
}

/****************************************************/

export type RootStackParamList = {
  Home: Card | undefined

  Register:
    | {
        card?: Card
      }
    | undefined

  RegisterConfirmation:
    | {
        card: Card
      }
    | undefined

  LoadingAnimation: undefined

  MyCards:
    | {
        allCards?: Card[]
      }
    | undefined
}

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}

export type RootStackScreenProps<ScreenName extends keyof RootStackParamList> =
  NativeStackScreenProps<RootStackParamList, ScreenName>
