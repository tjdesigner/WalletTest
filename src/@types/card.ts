export interface Card {
  id?: string
  cardName?: string
  cardNumber?: string
  name?: string
  expirationDate?: string
  cvv?: string
}
