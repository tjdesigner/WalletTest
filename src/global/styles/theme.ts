export default {
  colors: {
    black: "black",
    white: "white",
    danger: "#DC1637",
    info: "#12C2E9",
    warning: "#FFE900",
    success: "#2ec64e",
    background: "rgba(238, 238, 238, .2)",
    primary: "#142995",
    secondary: "#12C2E9",
    tertiary: "#A5FF32",
    backgroundDisabled: "#EEEEEE",
    textDisabled: "#BBBBBB",
  },

  fonts: {
    family: {
      regular: "PTSans-Regular",
      italic: "PTSans-Italic",
      boldItalic: "PTSans-BoldItalic",
      bold: "PTSans-Bold",
    },
    weight: {
      normal: "normal",
      bold: "bold",
      l1: "100",
      l2: "200",
      l3: "300",
      l4: "400",
      l5: "500",
      l6: "600",
      l7: "700",
      l8: "800",
      l9: "900",
      undefined,
    },
  },
  fontSize: {
    xxxs: `${1}px`,
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    default: `${16}px`,
    medium: `${24}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${60}px`,
    veryLarge: `${76}px`,
    titlePage: `${20}px`,
    title: `${28}px`,
    specific: `${18}px`,
  },
  fontSizeNumber: {
    xxxs: 1,
    xxs: 2,
    xs: 4,
    small: 8,
    default: 16,
    medium: 24,
    large: 32,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
    titlePage: 20,
    title: 28,
    specific: 18,
  },
  radius: {
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    small2: `${12}px`,
    default: `${16}px`,
    medium: `${20}px`,
    medium2: `${24}px`,
    medium3: `${28}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${54}px`,
    veryLarge: `${76}px`,
  },
  radiusNumber: {
    xxs: 2,
    xs: 4,
    small: 8,
    small2: 12,
    default: 16,
    medium: 24,
    large: 32,
    large2: 36,
    large3: 40,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
  },
  spaces: {
    xxs: `${2}px`,
    xs: `${4}px`,
    small: `${8}px`,
    default: `${16}px`,
    medium: `${20}px`,
    medium2: `${24}px`,
    medium3: `${28}px`,
    large: `${32}px`,
    large2: `${36}px`,
    large3: `${40}px`,
    xLarge: `${44}px`,
    xxLarge: `${48}px`,
    xxxLarge: `${54}px`,
    veryLarge: `${76}px`,
  },
  spacesNumber: {
    xxs: 2,
    xs: 4,
    small: 8,
    default: 16,
    medium: 24,
    large: 32,
    large2: 36,
    large3: 40,
    xLarge: 44,
    xxLarge: 48,
    xxxLarge: 60,
    veryLarge: 76,
  },
}
