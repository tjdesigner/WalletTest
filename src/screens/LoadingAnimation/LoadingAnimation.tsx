import React, { useEffect } from "react";
import { LoadingAnimationContainer, LoadingAnimationProps } from "./styles";
import { RootStackScreenProps } from "../../@types/navigation";
import { ActivityIndicator } from "react-native";
import theme from "../../global/styles/theme";

export const LoadingAnimation = ({ navigation, route }: RootStackScreenProps<'LoadingAnimation'>) => {

    useEffect(() => {
        setTimeout(() => {
            navigation.navigate('RegisterConfirmation')
        }, 3000);
    }, [])

    return (
        <LoadingAnimationContainer>
            <ActivityIndicator size={"large"} color={theme.colors.white} />
        </LoadingAnimationContainer >
    )
}
