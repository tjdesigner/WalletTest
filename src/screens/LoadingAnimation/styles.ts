import styled from "styled-components/native"
import { ViewProps } from "react-native"
import theme from "../../global/styles/theme"

export interface LoadingAnimationProps extends ViewProps {}

export const LoadingAnimationContainer = styled.View<LoadingAnimationProps>`
  flex: 1;
  background-color: ${theme.colors.primary};
`
