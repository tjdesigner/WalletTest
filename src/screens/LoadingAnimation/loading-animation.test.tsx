import React, { useState as useStateMock } from "react";
import { act, fireEvent, render } from '@testing-library/react-native'
import { RootStackParamList } from "../../@types/navigation";
import type { NativeStackScreenProps } from "@react-navigation/native-stack";
import { ThemeProvider } from "styled-components";
import theme from "../../global/styles/theme";
import dataBase from '../../../db.json'
import LoadingAnimation from "./LoadingAnimation";

const navigateMock = jest.fn()

type LoadingAnimationProps = NativeStackScreenProps<RootStackParamList, 'LoadingAnimation'>

const optionsNavigation = {
    navigation: {
        navigate: navigateMock
    }
} as unknown as LoadingAnimationProps

describe('MyCards', () => {

    it('should render mycards screen', () => {
        const { toJSON } = render(<ThemeProvider theme={theme}>
            <LoadingAnimation {...optionsNavigation} />
        </ThemeProvider>)
        expect(toJSON()).toMatchSnapshot()
    })

})
