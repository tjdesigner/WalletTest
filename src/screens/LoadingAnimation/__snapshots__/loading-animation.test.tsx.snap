// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCards should render mycards screen 1`] = `
<View
  style={
    {
      "backgroundColor": "#142995",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingTop": 17,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      animatedStyle={
        {
          "value": {
            "left": -900,
            "top": -3000,
          },
        }
      }
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(238, 238, 238, .2)",
          "borderRadius": 50,
          "height": 900,
          "left": -900,
          "position": "absolute",
          "top": -3000,
          "transform": [
            {
              "rotate": "55deg",
            },
          ],
          "width": 900,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 0,
          "justifyContent": "center",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "width": 0,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="50"
        bbWidth="50"
        fill="transparent"
        focusable={false}
        height={50}
        id="wallet"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 50,
              "width": 50,
            },
          ]
        }
        vbHeight={52}
        vbWidth={52}
        width={50}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 0,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M45.4444 51H6.55556C5.08213 51 3.66905 50.4147 2.62718 49.3728C1.58532 48.3309 1 46.9179 1 45.4444V6.55556C1 5.08213 1.58532 3.66905 2.62718 2.62718C3.66905 1.58532 5.08213 1 6.55556 1H45.4444C46.9179 1 48.3309 1.58532 49.3728 2.62718C50.4147 3.66905 51 5.08213 51 6.55556V45.4444C51 46.9179 50.4147 48.3309 49.3728 49.3728C48.3309 50.4147 46.9179 51 45.4444 51Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              {
                "payload": 4279419625,
                "type": 0,
              }
            }
            strokeWidth="2"
          />
          <RNSVGPath
            d="M1 34.3333H18.7778C19.6972 34.3333 20.4583 35.1056 20.8611 35.9333C21.4333 37.1083 22.7889 38.5 26 38.5C29.2111 38.5 30.5667 37.1111 31.1389 35.9333C31.5417 35.1056 32.3028 34.3333 33.2222 34.3333H51M1 12.1111H51M1 23.2222H51"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              {
                "payload": 4279419625,
                "type": 0,
              }
            }
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      animatedStyle={
        {
          "value": {
            "bottom": -3000,
            "right": -900,
          },
        }
      }
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(238, 238, 238, .2)",
          "borderRadius": 50,
          "bottom": -3000,
          "height": 900,
          "position": "absolute",
          "right": -900,
          "transform": [
            {
              "rotate": "55deg",
            },
          ],
          "width": 900,
        }
      }
    />
  </View>
</View>
`;
